source/source.go

```diff
@@ -114,6 +115,11 @@ func (s *Source) Teardown(ctx context.Context) error {
 	logger := sdk.Logger(ctx).With().Str("Class", "Source").Str("Method", "Teardown").Logger()
 	logger.Trace().Msg("Starting Teardown the Source Connector...")
 
+	if s.combinedIterator != nil {
+		s.combinedIterator.Stop()
+		s.combinedIterator = nil
```

Comment:
Why do we need to set it to nil?

Reply:
Intention behind this is to make combined iterator invalidate and should not be used further after the teardown.
Because if the read is called after teardown then it should return an error of "combined iterator is not initialized".

Change the source read also to support when combined iterator is nil.

Reply:
The SDK guarantees that Teardown() will be the last call to the plugin, so from that point of view, setting it to nil is not really needed. If a function is called after Teardown() then something is messed up in the SDK or Conduit.

Reply:
So making combinedIterator nil would be a proactive step even if something messed up with the SDK. So should i continue with this ??

Reply:
Got it. Let me remove it